src/backend/distributed/transaction/lock_graph.c

```diff
@@ -590,9 +641,13 @@ AddEdgesForWaitQueue(WaitGraph *waitGraph, PGPROC *waitingProc, PROCStack *remai
 	{
 		int awaitMask = LOCKBIT_ON(currentProc->waitLockMode);
 
-		/* skip processes from the same lock group and ones that don't conflict */
+		/*
+		 * Skip processes from the same lock group and ones that don't conflict
+		 * and waiting on the relation extension.
```

Comment:
-> Skip processes from the same lock group, processes that don't conflict, and processes that are waiting on a relation extension lock, which will be released shortly.

Reply:
Done